lsp-mode.el

```diff
@@ -3785,28 +3785,28 @@ interface TextDocumentEdit {
            (lsp-with-filename file-name
              (lsp--apply-text-edits (lsp:text-document-edit-edits edit))))))))
 
-(lsp-defun lsp--position-compare ((&Position :line left-line
```

Comment:
why did you change the convention from left/right to a/b? It is used here, and in 4+ packages in emacs-lsp organization. Also, is reversing the result of compare function neccessary?

Reply:
I mentioned it in my commit message:
```
I reused but tweaked lsp--position-compare:
 - Invert comparison so t is returned if the first param is less (I
   think this is more conventional).
 - Rename to lsp--position-before-p and make doc string clearer.
 - Rename args from "left" and "right" to "a" and "b" (left/right is
   confusing since it suggests a positional relationship).
```

"compare" gives no indication of how it compares the params. I read the doc string but it also did not indicate, so I had to examine the function. It turned out to be the opposite of what I would have guessed. To make it clearer, I renamed it `...less-p` indicating it tests if the first param is less than the second (and I flipped the logic to match). I think "less" is more conventional in programming than "more" (e.g. when you call "sort" by default it arranges in ascending order in most languages).

I found "left" and "right" confusing because it seemed like it was saying one of the positions was on the left (in the buffer), and one on the right. I switched to "a" and "b" since those do not suggest any sort of relationship between the two parameters.

Reply:
I understand that but when you do these changes, with no tests to prove that there is no change in the behaviour I have to go there and do manual testing for all of the changed positions. In this case, lsp--text-edit-sort-predicate should perform the stable sort and if it doesn't this will break applying changes. This is one of the critical sections in the application that has caused a lot of trouble in the past.

> I found "left" and "right" confusing because it seemed like it was saying one of the positions was on the left (in the buffer), and one on the right. I switched to "a" and "b" since those do not suggest any sort of relationship between the two parameters.

Left and right are pretty standard for sort related functions, please revert that.

Reply:
So there are no existing tests for the critical behavior? I figured if the tests passed I didn't break anything important. I can undo all the last changes.

Reply:
If the doc string for the function is not clear, you're welcome to change it. That would break nothing and you still get the desired behavior.

Reply:
Undid changes and clarified lsp--position-compare doc string.